compiler/optimizer/abstractinterpreter/IDT.cpp

```diff
@@ -39,6 +39,7 @@ void TR::IDT::print()
 
    if (!verboseInlining && !traceBIIDTGen)
       return;
+   TR_VerboseLog::vlogAcquire();
```

Comment:
Since we're now dynamically allocating memory for the buffers, we need to be concerned with exception safety. Allocation failure will throw an exception to fail the compilation, and if that exception propagates through this frame, we will fail to release the lock

Similarly, there is an early return path that fails to release the lock as well

Reply:
There is an added check inside the early return for releasing the lock if the verboseInlining is ture

Reply:
Releasing before early return is good, but this code will still fail to release the lock if an exception is thrown, which can happen if we hit the scratch space limit while dynamically allocating memory to hold the output. To ensure that the lock will be released even on exception, it's necessary to release it from the destructor of a local variable. I've opened #6244 to implement such a mechanism. If we suppose for the moment that it will be merged without major changes, then it will be possible to fix the problem by replacing direct use of `vlogAcquire()`/`vlogRelease()` with the following:
```
TR_VerboseLog::CriticalSection vlogLock(verboseInlining);
```

Reply:
#6244 has been merged, so `TR_VerboseLog::CriticalSection` is now available

Reply:
All `vlogAcquire()`/`vlogRelease()` has been replaced by `TR_VerboseLog::CriticalSection vlogLock(verboseInlining);` now.